### PR TITLE
fix: organization owner logic

### LINKED ISSUE #2441: check-allowed-organization! is broken
... it's looking at the user's organizations, instead of the organization's owners

### PR DESCRIPTION
**Pull request by opqdonut**
on top of #2440

fixes #2441

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue
- [x] note if PR is on top of other PR

## Documentation
- [x] update changelog if necessary

## Testing
- [x] complex logic is unit tested
- [x] valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [x] no critical TODOs left to implement

### CODE REVIEW

File: src/clj/rems/api/services/util.clj [hunk 1]
```diff
@@ -18,46 +17,39 @@
...
       (is (not (forbidden-organization? {:organization/id nil})))))
 
   (testing "for owner who is also an organization owner, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
-              context/*roles* #{:owner :organization-owner}]
+    (binding [context/*user* {:eppn "x"}
+              context/*roles* #{:owner}]
       (is (not (forbidden-organization? {:organization/id "own organization"})))
-      (is (not (forbidden-organization? {:organization/id "not own organization"})))
+      (is (not (forbidden-organization? {:organization/id "not own organization" :organization/owners [{:userid "x"}]})))
```
Comment (Macroz) [inline src/clj/rems/api/services/util.clj:44]: Could be made slightly more obvious if the id would be "organization owned by" or something?

File: src/clj/rems/api/services/util.clj [hunk 2]
```diff
@@ -18,46 +17,39 @@
...
-
-  (testing "for organization owner with no organization defined, all organizations are forbidden"
-    (binding [context/*user* {}
-              context/*roles* #{:organization-owner}]
-      (is (forbidden-organization? {:organization/id "own organization"}))
+    (binding [context/*user* {:eppn "x"}
+              context/*roles* #{}]
+      (is (not (forbidden-organization? {:organization/id "own organization" :organization/owners [{:userid "x"}]})))
+      (is (not (forbidden-organization? {:organization/id "other own organization" :organization/owners [{:userid "y"} {:userid "x"}]})))
+      (is (forbidden-organization? {:organization/id "not own organization" :organization/owners [{:userid "y"}]}))
```
Comment (Macroz) [inline src/clj/rems/api/services/util.clj:53]: This "y" could also be "unrelated-other-user"

**Review by Macroz — APPROVED** [review]
I would try to slightly improve the `forbidden-organization?` test data naming to improve the readability of the different cases but otherwise didn't notice anything strange.

File: src/clj/rems/api/services/util.clj [hunk 3]
```diff
@@ -1,65 +1,72 @@
...
+        (is (not (forbidden-organization? org-bob-carl)))))
 
-  (testing "for owner who is also an organization owner, all organizations are permitted"
-    (binding [context/*user* {:organizations [{:organization/id "own organization"}]}
-              context/*roles* #{:owner :organization-owner}]
-      (is (not (forbidden-organization? {:organization/id "own organization"})))
-      (is (not (forbidden-organization? {:organization/id "not own organization"})))
-      (is (not (forbidden-organization? {:organization/id ""})))
-      (is (not (forbidden-organization? {:organization/id nil})))))
+    (testing "for owner, all organizations are permitted"
```
Comment (Macroz) [inline src/clj/rems/api/services/util.clj:47]: This feels like duplicate of the previous. Should something change in the description or the content?

**Review by Macroz — COMMENTED** [review]
Looks better.

File: src/clj/rems/api/services/util.clj (hunk 3 above)
Comment (opqdonut) [reply src/clj/rems/api/services/util.clj]: woops, done

**Review by Macroz — APPROVED** [review]

**Pull request merged**